Gems/Atom/RHI/Code/Tests/PageTileAllocatorTests.cpp

```diff
@@ -0,0 +1,170 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+
+#include "RHITestFixture.h"
+#include <AzCore/UnitTest/UnitTest.h>
+#include <Atom/RHI/PageTileAllocator.h>
+#include <AzCore/Math/Random.h>
+
+namespace UnitTest
+{
+    using namespace AZ;
+
+    class PageTileAllocatorTest
+        : public RHITestFixture
+    {
+    public:
+        PageTileAllocatorTest()
+            : RHITestFixture()
+        {}
+
+        // Return true if there is no overlapping between all the input tile groups
+        bool ValidateTilesNotOverlap(const AZStd::vector<RHI::PageTileSpan>& tilesList)
+        {
+            AZStd::set<RHI::PageTileSpan, RHI::PageTileSpan::Compare> sortedTilesList;
+
+            for (const auto& tiles : tilesList)
+            {
+                sortedTilesList.insert(tiles);
+            }
+
+            uint32_t lastTile = 0;
+            for (const auto& tiles1 : sortedTilesList)
+            {
+                uint32_t newlastTile = tiles1.m_offset + tiles1.m_tileCount;
+                if (lastTile < newlastTile)
+                {
+                    lastTile = newlastTile;
+                }
+                else
+                {
+                    return false;
+                }
```

Comment:
I don't think this is detecting all overlap cases, it's only detecting cases where one span is entirely contained in another span. Shouldn't this also check lastTile against tiles1.m_offset?
```suggestion
                if(lastTile > tiles1.m_offset)
                {
                    return false;
                }
                
                uint32_t newlastTile = tiles1.m_offset + tiles1.m_tileCount;
                
                if (lastTile < newlastTile)
                {
                    lastTile = newlastTile;
                }
                else
                {
                    return false;
                }
```